projects/origin-release/golang-1.14/Dockerfile

```diff
@@ -0,0 +1,55 @@
+#
+# This is the image that controls the standard build environment for releasing OpenShift Origin.
+# It is responsible for performing a cross platform build of OpenShift.
+#
+# The standard name for this image is openshift/origin-release
+#
+FROM        centos:7
+
+ENV VERSION=1.14.2 \
+    GOCACHE=/go/.cache \
+    GOARM=5 \
+    GOPATH=/go \
+    GOROOT=/usr/local/go \
+    GOFLAGS='-mod=vendor' \
```

Comment:
From the [Go 1.14 Release Notes](https://golang.org/doc/go1.14#go-command):
> ### Go command
> #### Vendoring
>
> When the main module contains a top-level vendor directory and its go.mod file specifies go 1.14 or higher, the go command now defaults to -mod=vendor for operations that accept that flag. A new value for that flag, -mod=mod, causes the go command to instead load modules from the module cache (as when no vendor directory is present).
>
> When -mod=vendor is set (explicitly or by default), the go command now verifies that the main module's vendor/modules.txt file is consistent with its go.mod file.

@stevekuznetsov given the above, I wonder if it would be a good idea to leave this flag out? Without it, Go should now "do the correct thing" depending on whether the project being built vendors dependencies or not.

Reply:
@grdryn no it does not unless you bump the go.mod:
> and its go.mod file specifies go 1.14 or higher

Reply:
@alvaroaleman yes, that's true; although I'd assume that projects would update the go.mod to specify that if they were going to use this image. I guess there's no guarantee of that though.